### DISCUSSION TITLE
Create a solution file with SolutionFile

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/msbuild#6320] [Q&A] [answered]
Our codebase is large (around 250 projects, where we have ~10 main applications and the rest are library projects). We have only large solution file that contains all projects in order to debug and to perform refactoring. The problem is that VS gets very slow with this large solution and daily work is almost impossible. For many tasks it would be sufficient though to have smaller projects. We are using solution filters but also here the problem is also the maintenance of these filters.

Currently Visual Studio requires solution files in order to work correctly, where for our purpose regular project files would be mostly sufficient if the 2% a solution file is able to do would be included in a C# project (we are using the DependsOn) and MultipleStartup projects.

So the idea came up that we auto generate solution files for each project and include all dependencies in a solution file by parsing the projects and manually create solutions. We have done this by using the Microsoft.Build packages which work very good, but there is one thing we couldn't find out so far.

There is currently a SolutionFile but it needs to be physically available. In our case it's no available and we need to create it from scratch. One problem is that it's very hard to create the solution file because of the project type UIDs.

So the question is if there is way to create a solution file with some of your classes or if there is even a better way to do what we want to do?

**Comment by jeffkl** [comment, marked as answer]
Please try out https://github.com/microsoft/slngen/

**Reply by msedi** [reply]
@jeffkl: Thanks Jeff, I think this is exactly what I was looking for. The overview section seems like a copy of my description ;-)

**Reply by msedi** [reply]
@jeffkl: Just to make sure. Does it work with a .NET 5 console application? The nuget packages are complaining...
